config/kernel-config-defined.m4

```diff
@@ -55,6 +55,25 @@ AC_DEFUN([ZFS_AC_KERNEL_SRC_CONFIG_THREAD_SIZE], [
 	])
 ])
 
+dnl #
+dnl # Check configured KSTACK_PAGES * pagesize
+dnl #
+dnl # On FreeBSD we check these runtime settings, it's only expected to be false
+dnl # on 32b mips and arm with default configs.
+dnl #
+AC_DEFUN([ZFS_AC_KERNEL_CONFIG_THREAD_SIZE_FREEBSD], [
+    AC_MSG_CHECKING([whether kernel was built with 16K or larger stacks])
+	PAGE_SIZE=`sysctl -n hw.pagesize`
+	KSTACK_PAGES=`sysctl -n kern.kstack_pages`
+	KSTACK_SIZE=`expr $KSTACK_PAGES \* $PAGE_SIZE`
```

Comment:
Thank you for catching this, but you are checking variables of the running system, not a system you are building for, better represented by kernel headers.  For the mentioned mips and arm systems cross-building on x86 is quite popular, so your patch will be wrong. You should probably include sys/param.h and take KSTACK_PAGES and PAGE_SIZE from there.

Reply:
@amotin thanks have a look at this update